xonsh/prompt/job.py

```diff
@@ -1,14 +1,27 @@
 """Prompt formatter for current jobs"""
 
-import xonsh.jobs as xj
+import contextlib
+import typing as tp
+
+_current_cmds: tp.Optional[list] = None
+
+
+@contextlib.contextmanager
+def update_current_cmds(cmds):
+    """Context manager that updates the information used by _current_job()"""
+    global _current_cmds
```

Comment:
can this be converted to a PromptField class. That way we dont need to deal with globals ?

Reply:
Great idea! I'll try to do that.